src/OpenTelemetry/Trace/Decision.cs

```diff
@@ -0,0 +1,43 @@
+﻿// <copyright file="Decision.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Trace
+{
+    /// <summary>
+    /// Enumeration to define sampling decision.
+    /// </summary>
+    public enum Decision
```

Comment:
I'm concerned about having this name under `OpenTelemetry.Trace`:
1. `Decision` could be used by other namespaces given it is a common word.
2. It is not clear literally what does `OpenTelemetry.Trace.Decision` mean.

This is what [OpenTelemetry Python is doing](https://github.com/open-telemetry/opentelemetry-python/blob/5cb01d6c907948ba56ab4595c4a7d185fbd02980/opentelemetry-api/src/opentelemetry/trace/sampling.py#L69) and we might borrow some idea.

Reply:
My suggestion would be:
1. Introduce `OpenTelemetry.Trace.Sampling` namespace.
2. Put the `Sampler` base class, `Decision` and all the samplers under that namespace.
3. Folks who don't use sampler explicitly don't need to have `using OpenTelemetry.Trace.Sampling`.
4. Folks who use sampler or write sampler will need to have `using OpenTelemetry.Trace.Sampling`.

An alternative approach would be calling this class `SamplingDecision`, but that seems to be worse?

Reply:
Agree with @reyang it's a bit too generic to have in a common namespace. I think `SamplingDecision` is a good change to make regardless of where it goes. Decision alone is hyper-generic 🤣

Reply:
I would prefer `SamplingDecision` as the class name if we agree to move the current samplers from `src/Trace/Samplers` to `src/Trace` and remove the `OpenTelemetry.Trace.Samplers` namespace.

Reply:
Moved the current samplers from `src/Trace/Samplers` to `src/Trace` and removed the `OpenTelemetry.Trace.Samplers` namespace. Using `SamplingDecision` as the enum name.